src/client/Client.cc

```diff
@@ -10971,6 +10962,20 @@ int64_t Client::_read(Fh *f, int64_t offset, uint64_t size, bufferlist *bl,
     // branch below but in a non-blocking fashion. The code in _read_sync
     // is duplicated and modified and exists in
     // C_Read_Sync_NonBlocking::finish().
+
+    // trim read based on file size?
+    if ((offset >= in->size) || (size == 0)) {
```

Comment:
seeing gcc warnings from this:
```
ceph/src/client/Client.cc:10967:17: warning: comparison of integer expressions of different signedness: ‘int64_t’ {aka ‘long int’} and ‘uint64_t’ {aka ‘long unsigned int’} [-Wsign-compare]
10967 |     if ((offset >= in->size) || (size == 0)) {
      |          ~~~~~~~^~~~~~~~~~~
```
tracked in https://tracker.ceph.com/issues/68552